ring.go

```diff
@@ -213,32 +215,72 @@ type ringShards struct {
 
 	mu       sync.RWMutex
 	hash     ConsistentHash
-	shards   map[string]*ringShard // read only
-	list     []*ringShard          // read only
+	shards   map[string]*ringShard // read only, replaced by SetAddrs
+	list     []*ringShard          // read only, replaced by SetAddrs
 	numShard int
 	closed   bool
 }
 
 func newRingShards(opt *RingOptions) *ringShards {
-	shards := make(map[string]*ringShard, len(opt.Addrs))
+	shards, list := createRingShards(opt, opt.Addrs)
+
+	c := &ringShards{
+		opt: opt,
+
+		shards: shards,
+		list:   list,
+	}
+	c.rebalance()
```

Comment:
I think we now can change this to 
```go
c := &ringShards{opt: opt}
c.SetAddrs(opt.Addrs)

return c
```
and remove `createRingShards`.

Reply:
makes sense, done.